lib/model/settings.dart

```diff
@@ -45,6 +45,94 @@ enum BrowserPreference {
+
+  /// Describes a setting which enables an in-progress feature of the app.
+  ///
+  /// Sometimes when building a complex feature it's useful to merge PRs that
+  /// make partial progress, and then to have the feature's logic gated behind
+  /// a setting that serves as a "feature flag".
+  /// This enables those working on the feature to enable the flag in order to
+  /// see the current incomplete behavior, while for everyone else it remains
+  /// disabled and so (barring bugs in the use of the flag itself) has no effect.
+  ///
+  /// These settings are primarily meant for people developing Zulip to use,
+  /// and so appear in an out-of-the-way part of the settings UI.
+  ///
+  /// Settings of this kind are costly to the health of the codebase if
+  /// allowed to accumulate.  Most features don't need one, even features that
+  /// take two or three PRs to implement.  See discussion at:
+  ///   https://github.com/zulip/zulip-flutter/issues/1409#issuecomment-2725793787
+  /// When a feature flag is introduced, take care to drive the project to
+  /// completion, either by merge or removal, so that the flag can be retired
+  /// within a period of a few weeks or months.
+  experimentalFeatureFlag,
+  ;
+}
+
+/// A bool-valued, account-independent setting the user might set.
+///
+/// These are recorded in the table [BoolGlobalSettings].
+/// To read the value of one of these settings, use [GlobalSettingsStore.getBool];
+/// to set the value, use [GlobalSettingsStore.setBool].
+///
+/// To introduce a new setting, add a value to this enum.
+/// Avoid re-using any old names found in the "former settings" list.
+///
+/// To remove a setting, comment it out and move to the "former settings" list.
+/// Tracking the names of settings that formerly existed is important because
+/// they may still appear in users' databases, which means that if we were to
+/// accidentally reuse one for an unrelated new setting then users would
+/// unwittingly get those values applied to the new setting,
+/// which could cause very confusing buggy behavior.
+///
+/// (If the list of former settings gets long, we could do a migration to clear
+/// them from existing installs, and then drop the list.  We don't do that
+/// eagerly each time, to avoid creating a new schema version each time we
+/// finish an experimental feature.)
+enum BoolGlobalSetting {
+  /// A non-setting to ensure this enum has at least one value.
+  ///
+  /// Leave this in place even when there are experimental feature flags too.
+  /// That way when we remove those, this is already here.
+  /// (Having one stable value in this enum is also handy for tests.)
+  placeholderIgnore(GlobalSettingType.placeholder, false),
+
+  renderKatex(GlobalSettingType.experimentalFeatureFlag, false),
+
+  // Former settings which might exist in the database,
+  // whose names should therefore not be reused:
+  // (this list is empty so far)
+  ;
+
+  const BoolGlobalSetting(this.type, this.default_);
```

Comment:
nit: move this or the final fields set by it so that they are next to each other